valhalla/sif/dynamiccost.h

```diff
@@ -345,7 +345,9 @@ class DynamicCost {
         ((disallow_mask & kDisallowEndRestriction) && edge->end_restriction()) ||
         ((disallow_mask & kDisallowSimpleRestriction) && edge->restrictions()) ||
         ((disallow_mask & kDisallowShortcut) && edge->is_shortcut());
-    return accessible && !assumed_restricted && (edge->use() != baldr::Use::kConstruction);
+    return accessible && !assumed_restricted &&
+           (edge->use() != baldr::Use::kConstruction ||
+            (ignore_constructions_ && edge->use() == baldr::Use::kConstruction));
```

Comment:
i think you can simplify, if the first part is false we already know the use is construction
```suggestion
    return accessible && !assumed_restricted &&
           (edge->use() != baldr::Use::kConstruction || ignore_constructions_);
```

Reply:
That is indeed simpler and still correct.